### PR TITLE
Restart

### PR DESCRIPTION
**Pull request by yanganto**
base on #642

### CODE REVIEW

File: src/bin/kneard-mgr.rs [hunk 1]
```diff
@@ -251,59 +276,117 @@ async fn rollback(
 
 async fn proxy(proxy_args: &ProxyArgs, config: &Config) -> Result<()> {
     let hosts = filter_hosts(&proxy_args.host, &config.hosts)?;
+    if hosts.len() > 1 {
+        println!(
+            "proxy action only work on singly host, kneard-mgr will proxy {}",
```
Comment (Mic92) [inline src/bin/kneard-mgr.rs:281]:
```suggestion
            "Multiple hosts detected in configuration. Will proxy to machine: {}",
```

File: README.md [hunk 2]
```diff
@@ -60,22 +60,22 @@ Subcommand to run
...
-  maintenance-shutdown  Ask Kuutamod to schedule a shutdown in maintenance windows
-  ssh                   SSH into a host
-  help                  Print this message or the help of the given subcommand(s)
+  generate-config  Generate NixOS configuration
+  install          Install Validator on a given machine. This will remove all data of the current system!
+  dry-update       Upload update to host and show which actions would be performed on an update
+  update           Update validator
+  rollback         Rollback validator
+  proxy            Proxy remote rpc to local
+  restart          Schedule a restart in a maintenance window
```
Comment (Mic92) [inline README.md:69]:
```suggestion
  restart          Schedule a restart in a window where no block are expected to be produced by the validator
```

File: src/bin/kneard-mgr.rs [hunk 3]
```diff
@@ -123,12 +153,7 @@ enum Command {
     /// Proxy remote rpc to local
     Proxy(ProxyArgs),
     /// Schedule a restart in a maintenance window
```
Comment (Mic92) [inline src/bin/kneard-mgr.rs:155]:
```suggestion
    /// Schedule a restart in a window where no block are expected to be produced by the validator
```

File: README.md (hunk 2 above)
Comment (Mic92) [reply README.md:69]: "maintainance window" is a concept we made up but outsider will not know.

Comment (paulmattei) [reply README.md:69]: ..."where no blocks 'or chunks' are expected..."

File: src/bin/kneard-mgr.rs (hunk 3 above)
Comment (paulmattei) [reply src/bin/kneard-mgr.rs:155]: "...where no blocks 'or chunks' are expected..."

**Comment by yanganto** [comment]
Thanks for review.

File: README.md [hunk 4]
```diff
@@ -66,7 +66,7 @@ Commands:
   update           Update validator
   rollback         Rollback validator
   proxy            Proxy remote rpc to local
-  restart          Schedule a restart in a maintenance window
+  restart          Schedule a restart in a window where no blocks 'or chunks' are expected to be produced by the validator
```
Comment (paulmattei) [inline README.md:69]:
```suggestion
  restart          Schedule a restart in a window where no blocks or chunks are expected to be produced by the validator
```

File: src/bin/kneard-mgr.rs [hunk 5]
```diff
@@ -152,7 +152,7 @@ enum Command {
     Rollback(RollbackArgs),
     /// Proxy remote rpc to local
     Proxy(ProxyArgs),
-    /// Schedule a restart in a maintenance window
+    /// Schedule a restart in a window where no blocks 'or chunks' are expected to be produced by the validator
```
Comment (paulmattei) [inline src/bin/kneard-mgr.rs:155]:
```suggestion
    /// Schedule a restart in a window where no blocks or chunks are expected to be produced by the validator
```

**Review by Mic92 — APPROVED** [review]

**Comment by Mic92** [comment]
bors merge

**Comment by yanganto** [comment]
bors merge

**Pull request merged**